com.trollworks.gcs/src/com/trollworks/gcs/menu/item/OpenPageReferenceCommand.java

```diff
@@ -117,10 +117,25 @@ public static void openReference(String reference, String highlight) {
                 }
             }
             if (ref != null) {
+                //If the user has specified a custom PDF viewer string, use that
+                String pdfViewerString = prefs.getPdfViewerString();
+                int offsetPage = page + ref.getPageToIndexOffset();
+                boolean useCustomString = !pdfViewerString.isEmpty() && !pdfViewerString.isBlank();
                 try {
-                    PDFServer.showPDF(ref.getPath(), page + ref.getPageToIndexOffset());
+                    if(useCustomString)
+                    {
+                        String modifiedString = pdfViewerString
+                                .replace("%f",ref.getPath().toString())
+                                .replace("%p", Integer.toString(offsetPage));
+
+                        Runtime rt = Runtime.getRuntime();
+                            Process pr = rt.exec(modifiedString);
```

Comment:
I really don't like the "array of arguments" approach, but the only question I have left on implementing it is whether an argument and the value it might contain should go into one array entry. So the config would have a path to an executable, then a list of text strings, but would that list look like:
`[--file c:\GURPS-Characters.pdf]`, `[--page 5]`
or would parameters for arguments be their own entry in the list? Like:
`[--file]`, `[c:\GURPS-Characters.pdf]`, `[--page]`, `[5]`

Reply:
The latter. Each string you would expect to be in the array of args passed to main() in a C program should be in a separate string.

Reply:
That way, there is zero ambiguity about what the argument is. I'd have to test again to see if Java already does the escaping for you now when you use the arg list approach with the Process class -- a couple decades ago when I first ran into this on Windows, it did not, so you had to escape/quote them before putting them into the list that is passed to Process, while on all other OSs, you can just pass the list verbatim.